### PR TITLE
Pre-load vulnerabilities: NR-108350

### PR DESCRIPTION
**Pull request by reneelpetit**

### CODE REVIEW

File: shared/ui/Stream/SecurityIssuesWrapper.tsx [hunk 1]
```diff
@@ -322,6 +323,15 @@ export const SecurityIssuesWrapper = React.memo((props: Props) => {
 		[error]
 	);
 
+	if (data && data.totalRecords > 0) {
+		props.setVal(true);
+	}
+
+	const warningTooltip =
+		data && data.totalRecords > 0 && data.totalRecords === 1
```
Comment (bcanzanella) [inline shared/ui/Stream/SecurityIssuesWrapper.tsx:331]: can be simplified to `data && data.totalRecords === 1`

Comment (reneelpetit) [reply shared/ui/Stream/SecurityIssuesWrapper.tsx:331]: oh yeah good point!

File: shared/ui/Stream/SecurityIssuesWrapper.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ interface Props {
 	currentRepoId: string;
 	entityGuid: string;
 	accountId: number;
+	setVal: Function;
```
Comment (bcanzanella) [inline shared/ui/Stream/SecurityIssuesWrapper.tsx:33]: nitpick on the name, but what does "setVal" mean here? seems perhaps too generic?

Comment (reneelpetit) [reply shared/ui/Stream/SecurityIssuesWrapper.tsx:33]: yes good point, I'll make it less generic and more meaningful

File: shared/ui/Stream/SecurityIssuesWrapper.tsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ interface Props {
 	currentRepoId: string;
 	entityGuid: string;
 	accountId: number;
+	setHasVulnerabilities: Function;
```
Comment (purplexa) [inline shared/ui/Stream/SecurityIssuesWrapper.tsx:33]: probably worth being more specific and setting the type to `(value: boolean) => void`

File: shared/ui/Stream/SecurityIssuesWrapper.tsx [hunk 4]
```diff
@@ -322,6 +323,13 @@ export const SecurityIssuesWrapper = React.memo((props: Props) => {
 		[error]
 	);
 
+	if (data && data.totalRecords > 0) {
+		props.setHasVulnerabilities(true);
```
Comment (purplexa) [inline shared/ui/Stream/SecurityIssuesWrapper.tsx:327]: If I'm reading the docs right, I don't think you're supposed to call state setters directly during the render, at least not in a component other than the one where the state was defined. I think you're supposed to wrap this in a useEffect, like

```typescript
useEffect(() => {
	if (data && data.totalRecords > 0) {
		props.setHasVulnerabilities(true);
	}
}, [data, props.setHasVulnerabilities]);
```

**Review by purplexa — APPROVED** [review]

**Pull request merged**